etc/docker/ubuntu/install_deps.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```

Comment:
This is much neater! Can we do bash strict mode so a failure will kill the build?

Reply:
Oh absolutely, I just wasn't thinking about it.

Reply:
Done, I'll push it up once we get some results back from the currently enqueued CodeQL run so I can see if anything is still breaking.  Hitting some delays on getting the jobs picked up.

Reply:
Sounds good! My experience with codeQL (I've just used for Python) is that it's fairly slow. I have it on a few scattered projects and it has never hit any issues so I'm undecided about if I like it / if it's worth it - if it's not any slower than the CI here it shouldn't add time, but if it extends the running time we might want to consider just the scheduled run, or similar.

Reply:
Could be, I'm hoping it will be an adequate replacement for LGTM, since it's literally the same code query system by the same team under the hood.